rhscl-postgresql-apb/roles/rhscl-postgresql-apb-openshift/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+- name: create namespace
+  openshift_v1_project:
+    name: '{{ namespace }}'
+
+- name: create volumes
+  k8s_v1_persistent_volume_claim:
+    name: postgresql
+    namespace: '{{ namespace }}'
+    state: present
+    access_modes:
+      - ReadWriteOnce
+    resources_requests:
+      storage: 1Gi
+
+- name: create service
+  k8s_v1_service:
+    name: postgresql
+    namespace: '{{ namespace }}'
+    state: present
+    labels:
+      app: postgresql-demo-apb
+      service: postgresql
+    selector:
+      app: postgresql-demo-apb
+      service: postgresql
+    ports:
+    - name: port-5432
+      port: 5432
+      protocol: TCP
+      target_port: 5432
+  register: postgres_service
+
+- name: generate password
+  shell: pwgen -c -n 16 1
```

Comment:
I'd like a better way to do this, either with ansible directly or possibly even a better shell command.

The problem I hit here is that I wanted to add special characters, but the regex for the container (and possibly psql itself) does not allow quotes in the password and one of the first attempts I made included a ' character and the deployment failed. Aside from saying yes or no to special characters I have no control over what characters are included. Whether it comes down to a useful feature of ansible or a command line utility that provides finer controller I'd be open to suggestions.